src/main.rs

```diff
@@ -127,11 +182,44 @@ If none is set the normal git config is tried in the following order:
 Local repository config
 User config
 Global config");
-            process::exit(1);
-        }
-    };
+                process::exit(1);
+            }
+        };
+
+        cb.signature(signature.to_owned());
+    }
+
+    // In case we are in write-mode AND release mode,
+    // we will make sure we got all configuration settings
+    if !is_dry_run && release_mode {
+        let remote_url = match repo.find_remote("origin") {
+            Err(e) => print_exit!("Could not determine the origin remote url: {:?}", e),
+            Ok(remote) => {
+                let url = remote.url().expect("Remote URL is not valid UTF-8");
+                Url::parse(&url).expect("Remote URL can't be parsed")
+            }
+        };
+
+        let (user, repo_name) = user_repo_from_url(remote_url)
+            .unwrap_or_else(|e| print_exit!("Could not extract user and repository name from URL: {:?}", e));
+        cb.user(user);
+        cb.repository_name(repo_name);
+
+        let gh_token = env::var("GH_TOKEN")
+            .unwrap_or_else(|err| print_exit!("GH_TOKEN not set: {:?}", err));
+
+        let cargo_token = env::var("CARGO_TOKEN")
+            .unwrap_or_else(|err| print_exit!("CARGO_TOKEN not set: {:?}", err));
 
-    let version = toml_file::read_from_file(repository_path)
+        cb.gh_token(gh_token);
+        cb.cargo_token(cargo_token);
+    }
+
+    cb.repository(repo);
+    cb.branch("master".into());
```

Comment:
Actually we have a flag for that. This is a left-over from testing. Will use `args.flag_branch` instead (which defaults to master)